test/sql/upsert/minimal_reproducable_example.test

```diff
@@ -0,0 +1,42 @@
+# name: test/sql/upsert/minimal_reproducable_example.test
+# group: [upsert]
+
+statement ok
+pragma enable_verification;
+
+foreach parallel false true
+
+statement ok
+SET preserve_insertion_order=${parallel};
+
+# Create a table with 3 columns, 2 of which are indexed on
+statement ok
+create or replace table tbl(
```

Comment:
SQLite also supports `INSERT OR [IGNORE|REPLACE|...]` [short-cuts](https://www.sqlite.org/lang_insert.html). Could we support those as well?

Reply:
`INSERT OR IGNORE` is equivalent to `ON CONFLICT DO NOTHING`, that should be easy
`INSERT OR REPLACE` is equivalent to `ON CONFLICT DO UPDATE SET col1 = excluded.col1, col2 = excluded.col2`
Which requires some bind time magic, but I think we can make that work 👍 

SQLite says they use `REPLACE` as an alias for `INSERT OR REPLACE` for compatibility with MySQL, do we to also support this? Or do we want to use `UPSERT` as an alias for this?

Or we could even support both aliases?

Reply:
For now just supporting SQLite's aliases are fine - we can add more later